### PR TITLE
feat(docs): Adding guides and examples for dialogues

### PR DESCRIPTION
**Pull request by abhifetch**

### CODE REVIEW

File: pages/guides/dialogues/dialogues.mdx [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Dialogues 💬
```
Comment (devjsc) [inline pages/guides/dialogues/dialogues.mdx]: Should this be under uAgents?

**Review by devjsc — CHANGES_REQUESTED** [review]

File: pages/guides/dialogues/dialogues.mdx (hunk 1 above)
Comment (abhifetch) [reply pages/guides/dialogues/dialogues.mdx]: fixed

**Review by Dacksus — COMMENTED** [review]
The examples already do a decent job of explaining the chit chat example. The guide and concepts parts need another iteration to be more purposive / target-audience-oriented.

First, I would advice to add an "experimental" disclaimer, noting that this feature is under heavy development, no backwards compatibility is guaranteed, yet, and each update could introduce breaking changes. We can remove this, once the feature moves from the experimental path into the core framework.

The following dimensions or angles to dialogues played decisive roles,  which I'll try to explain to some extend afterwards:
- reusability of communication patterns (2-stage workflow of patterns -> dialogues)
- differentiation between "application developers" (implementing concrete use cases with agents) that need guidance and ease of use and "core developers" (providing infrastructure, enabling ecosystems) that need the full power of the framework to do the groundwork for application developers
- need for stateful communication (access communication history, parallel sessions, non-repudiation)
- need for abstract description of communication patterns to support better anticipation and evaluation of agents and their objectives for the ai engine
- the chit chat example is really just the simplest (and not a very useful) example to demonstrate the principle of dialogues and not meant as a standalone 'system'

Dialogues are not a self-containing feature and should be introduced and motivated in the right context as a consequential extension to protocols. This should also help understand why their usage is very similar to protocols (decorated message handlers) and why there are differences.
Protocols define clear interaction patterns, but without any state or sequence enforcement. They provide maximum flexibility but also put full responsibility on the developer (e.g., to think about all possible corner cases in any communication). In practice, we expect that the vast majority of interactions can be categorized by a handful of patterns.
Describing these general patterns is the first step for dialogues and consists of defining every possible flow / branch of a communication as a graph (nodes and edges). The main difference on this level to protocols is that a pattern defines a clear entry point, whereas in a protocol you must expect each message at any time. The resulting pattern definition can be instantiated with any message models, allowing adaptation to specific use cases with different content, yet following the same principles (e.g., different booking scenarios).
Even though it's not perfectly distinguished in the code base, I recommend using the terms 'pattern' for the general graph-based definition and 'dialogue' for a concrete implementation/instantiation. It might also make sense to clearly communicate the intended audience, e.g., for simply applying dialogues, I don't need to fully understand the way patterns are implemented

At this stage, a pattern takes the burden off an application developer to think about the communication structure, but they still need to completely implement every message handler. For further convenience we added the possibility to also add predefined message handlers to a pattern, to handle standard behavior such that as an application developer I only need to add/overwrite those handlers specific to my use case. We distinguish even more between behavior that can be replaced (message handlers) and behavior that is guaranteed to be executed in addition to whatever a developer will specify. This is the most important difference between the open and the predefined chitchat example, which should be highlighted and explained accordingly.

Beyond these conceptual things, it might be useful to put a bit more emphasis on the few apis dialogues expose to the developer:
- start_dialogue as a more explicit way to start a communication compared to protocols
- get_conversation to access past messages (relevant in more complex scenarios) and which also allows to handle multiple parallel dialogues with different agents (e.g., handle parallel ongoing booking requests where the agent manages a limited set of resources based on the active sessions)

At this point I would refrain from generally associating dialogues with DeltaV or chatbots.

File: pages/guides/agents/dialogues.mdx
Comment (Dacksus) [inline pages/guides/agents/dialogues.mdx:1]: As mentioned in the general comment, there is no such thing as a "chit-chat dialogue system". Chit-chat is an example already described in the examples above. This conceptual section should focus on the abstract motivation and interfaces of dialogues only referring to examples where necessary

File: pages/guides/agents/dialogues.mdx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    - **Message exchange:** Agents exchange message with each other, leading to change in state (Node) of dialogue forward.
+    - **State management:** The dialogue manager updates the dialogue state (Node) based on received messages and current state.
+
+
+## Dialogue Components:
+
+The `ChitChatDialogue` system is defined through a series of dialogue components that dictate how agents interact. Here are the key components:
+
+### Node (Dialogue State):
```
Comment (Dacksus) [inline pages/guides/agents/dialogues.mdx:42]: rather abstract on the level of a graph consisting of nodes and edges (possibly using the graph-like image you did for chit-chat as an example)

File: pages/guides/agents/dialogues.mdx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    - [**Dialogue Components:**](/guides/dialogues/dialogues#dialogue-components) Details on dialogue states, transitions, and message models.
+    - [**Setup and Configuration:**](/guides/dialogues/dialogues#setup-and-configuration) Instructions for setting up and running the agents.
+    - [**Implementation and Customization:**](/guides/dialogues/dialogues#implementation-and-customization) How to implement and customize dialogue flows.
+    - [**Examples:**](/guides/dialogues/dialogues#examples) Demonstrating practical implementation of dialogues.
+
+## System Architecture 📐
+
+The `ChitChatDialogue` is structured around network of agents that communicate through structured dialogues, Below is overview of architecture :
+
+### Components:
```
Comment (Dacksus) [inline pages/guides/agents/dialogues.mdx:25]: I'd remove agents and only focus on the distinction between patterns and dialogues as outlined in my general comment

File: pages/examples/predefined-dialogue-chitchat.mdx [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+```
+
+    - In the provided code, specific message types associated with different dialogue states trigger predefined functions that manage the flow and actions of the dialogue, automating the transition between states and handling interactions within the dialogue system efficiently.
+
+```python
+init_session.set_message_handler(InitiateChitChatDialogue, start_chitchat)
+start_dialogue.set_message_handler(AcceptChitChatDialogue, accept_chitchat)
+cont_dialogue.set_message_handler(ChitChatDialogueMessage, default)
+cont_dialogue.set_edge_handler(ChitChatDialogueMessage, persisting_function)
+end_session.set_message_handler(ConcludeChitChatDialogue, conclude_chitchat)
```
Comment (Dacksus) [inline pages/examples/predefined-dialogue-chitchat.mdx:158]: this is the most important part to explain and differentiate from the open example (also emphasize the difference between set_message_handler and set_edge_handler

**Pull request merged**